### PR TITLE
create categories endpoint for get requests

### PR DESCRIPTION
**Pull request by GiancarloFusiello**
In reference to https://github.com/libero/publisher/issues/245

- list categories using `/categories`
- filter categories by name using query parameters `/categories?name=<category-name>`
- get a single category by id `/categories/<id>`

All responses (including errors) are returned in JSON format.

### CODE REVIEW

File: scholarly_content_detail/error_handlers.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from flask import json
+
+
+def http_error_handler(error):
+    """Return JSON instead of HTML for HTTP errors."""
+    # start with the correct headers and status code from the error
+    response = error.get_response()
+    # replace the body with JSON
```
Comment (giorgiosironi) [inline scholarly_content_detail/error_handlers.py:8]: would follow https://tools.ietf.org/html/rfc7807 for this error response, namely `type`, `title`, `detail` for the 3 fields respectively

File: scholarly_content_detail/error_handlers.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    response.data = json.dumps({
+        "code": error.code,
+        "name": error.name,
+        "description": error.description,
+    })
+    response.content_type = "application/json"
```
Comment (giorgiosironi) [inline scholarly_content_detail/error_handlers.py]: would then be `application/problem+json`

File: tests/test_category_views.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    article = ArticleFactory()
+    category = CategoryFactory(articles=[article])
+    response = client.get('/categories/%s' % category.id)
+
+    assert response.status_code == 200
+    assert response.content_type == 'application/json'
+    expected = {
+        'id': category.id,
+        'name': category.name,
+        'articles': [
```
Comment (giorgiosironi) [inline tests/test_category_views.py]: would omit `articles` here for now, unless it's strictly needed by `jats-ingester`; I understand it's part of the default serialization but you I see no hard and fast rule for omitting or including the various relationships the `Category` is linked to inside the relational database.

File: scholarly_content_detail/error_handlers.py (hunk 1 above)
Comment (GiancarloFusiello) [reply scholarly_content_detail/error_handlers.py:8]: After having to link to three different documents for just the `title`; TL;DR.

Can you provide an example please?

**Review by giorgiosironi — COMMENTED** [review]
Apart from a minor comment on limiting the number of fields returned, the API looks like a good prototype that can be used from jats-ingester. We should be aware that since we have no shared standard defined at the product level, there may be changes required later (field names, more specific content types, etc.) and we should be comfortable making them.

File: scholarly_content_detail/error_handlers.py (hunk 2 above)
Comment (GiancarloFusiello) [reply scholarly_content_detail/error_handlers.py]: updated

File: tests/test_category_views.py (hunk 3 above)
Comment (GiancarloFusiello) [reply tests/test_category_views.py]: updated.

File: scholarly_content_detail/error_handlers.py (hunk 1 above)
Comment (giorgiosironi) [reply scholarly_content_detail/error_handlers.py:8]:
```
    response.data = json.dumps({
        "type": error.code,
        "title": error.name,
        "detail": error.description,
    })
```

**Review by giorgiosironi — APPROVED** [review]
Only renaming of fields in the error response remains

File: scholarly_content_detail/error_handlers.py (hunk 1 above)
Comment (GiancarloFusiello) [reply scholarly_content_detail/error_handlers.py:8]: updated.

**Pull request merged**